src/sparseml/core/session.py

```diff
@@ -46,26 +56,53 @@ class _CallbackContainer:
 
 
 class SparseSession:
+    """
+    A session for sparsification that holds the current state of the sparsification
+    """
+
     def __init__(self):
         self._lifecycle = SparsificationLifecycle()
 
     @property
     def lifecycle(self) -> SparsificationLifecycle:
+        """
+        :return: the lifecycle for the session
+        """
         return self._lifecycle
 
     @property
     def state(self) -> State:
+        """
+        :return: the current state of the session
+        """
```

Comment:
clarify that the state stores all the variables needed to perform the sparsification such as the recipe, model and data

Reply:
Done!